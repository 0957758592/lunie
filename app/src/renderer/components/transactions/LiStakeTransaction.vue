<template>
  <li-transaction
    :color="`#512584`"
    :time="transaction.time"
    :block="transaction.height"
  >
    <template v-if="txType === `cosmos-sdk/MsgCreateValidator`">
      <div slot="caption">
        Create validator&nbsp;<b>{{ atoms(tx.value.amount) }}</b><span>&nbsp;{{ tx.value.denom }}s</span>
      </div>
      <div slot="details">
        Moniker:&nbsp;<router-link :to="url + '/' + tx.validator_address">
          {{
            moniker(tx.validator_address)
          }}
        </router-link>
      </div>
    </template>
    <template v-else-if="txType === `cosmos-sdk/MsgEditValidator`">
      <div slot="caption">
        Edit validator&nbsp;<router-link :to="url + '/' + tx.validator_address">
          {{
            moniker(tx.validator_address)
          }}
        </router-link>
      </div>
    </template>
    <template v-else-if="txType === `cosmos-sdk/MsgDelegate`">
      <div slot="caption">
        Delegation&nbsp;<b>{{ atoms(tx.value.amount) }}</b><span>&nbsp;{{ tx.value.denom }}s</span>
      </div>
      <div slot="details">
        To&nbsp;<router-link :to="url + '/' + tx.validator_addr">
          {{
            moniker(tx.validator_addr)
          }}
        </router-link>
      </div>
    </template>
    <template v-else-if="txType === `cosmos-sdk/BeginRedelegate`">
      <div slot="caption">
        Redelegation&nbsp;<template>
          <b>
            {{
              calculatePrettifiedTokens(tx.validator_src_addr, tx.shares_amount)
            }}
          </b><span>&nbsp;{{ bondingDenom }}s</span>
        </template>
      </div>
      <div slot="details">
        From&nbsp;<router-link :to="url + '/' + tx.validator_src_addr">
          {{
            moniker(tx.validator_src_addr)
          }}
        </router-link>
        to&nbsp;<router-link :to="url + '/' + tx.validator_dst_addr">
          {{
            moniker(tx.validator_dst_addr)
          }}
        </router-link>
      </div>
    </template>
    <template v-else-if="txType === `cosmos-sdk/Undelegate`">
      <div slot="caption">
        Undelegation&nbsp;<template>
          <b>
            {{
              calculatePrettifiedTokens(tx.validator_addr, tx.shares_amount)
            }}
          </b><span>&nbsp;{{ bondingDenom }}s</span>
        </template><template
          v-if="timeDiff"
        >
          <span
            class="tx-unbonding__time-diff"
          >
            &nbsp;{{ timeDiff }}
          </span>
        </template>
      </div>
      <div slot="details">
        From&nbsp;<router-link :to="url + '/' + tx.validator_addr">
          {{
            moniker(tx.validator_addr)
          }}
        </router-link>
      </div>
    </template>
    <template v-else-if="txType === `cosmos-sdk/MsgUnjail`">
      <div slot="caption">
        Unjail&nbsp;<router-link :to="url + '/' + tx.address">
          {{
            moniker(tx.address)
          }}
        </router-link>
      </div>
    </template>
  </li-transaction>
</template>

<script>
import LiTransaction from "./LiTransaction"
import { pretty, atoms } from "../../scripts/num.js"
import { calculateTokens } from "../../scripts/common.js"
import moment from "moment"

/*
 * undelegation tx need a preprocessing, where shares are translated into transaction.balance: {amount, denom}
 */

export default {
	name: `li-stake-transaction`,
	components: { LiTransaction },
	props: {
		transaction: {
			type: Object,
			required: true
		},
		validators: {
			type: Array,
			required: true
		},
		url: {
			type: String,
			required: true
		},
		bondingDenom: {
			type: String,
			required: true
		},
		unbondingTime: {
			type: Number,
			default: null
		},
		txType: {
			type: String,
			required: true
		}
	},
	data: () => ({
		atoms,
		pretty
	}),
	computed: {
		tx() {
			return this.transaction.tx.value.msg[0].value
		},
		timeDiff() {
			// only show time diff if still waiting to be terminated
			if (this.state !== `locked`) return ``

			return `(liquid ` + moment(this.unbondingTime).fromNow() + `)`
		},
		// unbonding transactions can be in the state 'locked', 'ended'
		// the transaction needs to be enriched from the outside with `unbondingDelegation`
		state() {
			if (!this.unbondingTime) return `ended`
			if (this.unbondingTime - Date.now() <= 0) return `ended`
			return `locked`
		}
	},
	methods: {
		moniker(validatorAddr) {
			const validator = this.validators.find(
				c => c.operator_address === validatorAddr
			)
			return validator ? validator.description.moniker : validatorAddr
		},
		calculatePrettifiedTokens(validatorAddr, shares) {
			const validator = this.validators.find(
				val => val.operator_address === validatorAddr
			)
			return this.atoms(calculateTokens(validator, shares).toNumber())
		}
	}
}
</script>

<style>
.tm-li-tx {
	display: flex;
	font-size: sm;
}

.tm-li-tx .tx-icon {
	padding: 0 0.5rem;
	background: var(--app-fg);
	display: flex;
	align-items: center;
	justify-content: center;
}

.tm-li-tx .tx-container {
	flex-direction: column;
	flex-wrap: nowrap;
	padding: 0.5rem 0;
	margin: 0.5rem 0;
	display: flex;
	width: 100%;
	min-width: 0;
}

.tm-li-tx .tx-element {
	padding: 0 2rem 0 1.5rem;
	line-height: 1.5rem;
}

.tm-li-tx .tx-coin .value {
	flex: 0 0 100%;
	font-size: sm;
	color: var(--dim);
}

.tm-li-tx .tx-coin .value::before {
	content: "";
	display: inline;
}

.tm-li-tx .tx-coin .key {
	font-weight: 500;
	font-size: m;
}

.tm-li-tx .tx-coin .value,
.tm-li-tx .tx-coin .key {
	line-height: 1.5rem;
}

.tm-li-tx .tx-address {
	white-space: nowrap;
	overflow: hidden;
	text-overflow: ellipsis;
	color: var(--dim);
	font-size: sm;
}

.tm-li-tx.tm-li-tx-sent .tx-coin .value::before {
	content: "-";
}

.tm-li-tx.tm-li-tx-received .tx-icon {
	background: var(--app-fg);
}

.tm-li-tx.tm-li-tx-received .tx-coin .value {
	color: success;
}

.tm-li-tx.tm-li-tx-received .tx-coin .value::before {
	content: "+";
}

.tm-li-tx:hover {
	cursor: pointer;
}

.tx-unbonding__time-diff {
	font-size: var(--sm);
}

@media screen and (min-width: 700px) {
	.tm-li-tx {
		font-size: 0.875rem;
	}

	.tm-li-tx .tx-container {
		flex-direction: row;
	}

	.tm-li-tx .tx-container .tx-coins {
		flex: 0 0 9rem;
		padding: 0;
		min-width: 0;
	}

	.tm-li-tx .tx-container .tx-coins .tx-coin {
		padding: 0 1.5rem 0;
	}

	.tm-li-tx .tx-container .tx-coins .tx-coin .key {
		white-space: nowrap;
		overflow: hidden;
		text-overflow: ellipsis;
	}
}
</style>
