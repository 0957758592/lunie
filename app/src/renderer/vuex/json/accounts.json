{
  "difbruba": {
    "key": "PubKeyExample",
    "sequence": 1001,
    "coins": [
      { "denom": "eth", "amount": 445093 },
      { "denom": "btc", "amount": 82244 }
    ]
  },
  "q3ifnbe9": {
    "key": "PubKeyExample",
    "sequence": 1000,
    "coins": [
      { "denom": "atom", "amount": 8444 },
      { "denom": "btc", "amount": 32.423423 }
    ]
  },
  "e0gneidf": {
    "key": "PubKeyExample",
    "sequence": 999,
    "coins": [
      { "denom": "atom", "amount": 1000 },
      { "denom": "eth", "amount": 104.294 }
    ]
  },
  "difbfodb": {
    "key": "PubKeyExample",
    "sequence": 998,
    "coins": [
      { "denom": "atom", "amount": 10349 },
      { "denom": "eth", "amount": 2043.23423 },
      { "denom": "btc", "amount": 52.321 }
    ]
  }
}
