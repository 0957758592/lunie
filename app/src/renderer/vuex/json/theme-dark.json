{
  "app-fg": "hsl(233, 43%, 10%)",
  "app-bg": "hsl(233, 37%, 18%)",
  "app-bg-light": "hsl(233, 40%, 14%)",
  "app-bg-alpha": "hsla(237, 18%, 20%, 95%)",
  "txt": "hsl(0, 0%, 100%)",
  "dim": "hsl(240, 5%, 75%)",
  "bc": "hsl(233, 22%, 23%)",
  "bc-dim": "hsl(233, 33%, 16%)",
  "bc-light": "#dfdfdf",
  "box-shadow": "hsla(0,0,0,0.2)",
  "hover-bg": "hsla(233, 88%, 90%, 0.2)",
  "input-bc": "hsl(233, 22%, 33%)",
  "input-bc-hover": "hsl(233, 22%, 40%)",
  "link": "hsl(233, 88%, 82%)"
}
