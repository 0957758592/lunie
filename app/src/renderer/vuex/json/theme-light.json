{
  "app-fg": "hsl(0,0%,92%)",
  "app-bg": "#f5f5f5",
  "app-bg-light": "#fff",
  "app-bg-alpha": "hsla(233, 0%, 0%, 5%)",
  "txt": "#000",
  "dim": "#3b3b3b",
  "bc": "#ddd",
  "bc-dim": "#cdcbcb",
  "bc-light": "#dfdfdf",
  "box-shadow": "hsla(0,0,80%,0.2)",
  "hover-bg": "hsla(233, 88%, 90%, 0.2)",
  "input-bc": "hsl(233, 22%, 67%)",
  "input-bc-hover": "hsl(233, 22%, 40%)",
  "link": "hsl(233, 88%, 57%)"
}
