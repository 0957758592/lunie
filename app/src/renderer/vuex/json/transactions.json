{
  "dri3bdow": {
    "gas": 114,
    "fee": { "denom": "test", "amount": 125 },
    "inputs": [
      {
        "address": "Buffer",
        "coins": [
          { "denom": "test", "amount": 1000000 },
          { "denom": "atom", "amount": 4932 }
        ],
        "sequence": 1,
        "signature": "Buffer",
        "pubkey": "PubKey"
       }
    ],
    "outputs": [
      {
        "address": "Buffer",
        "coins": [
          { "denom": "test", "amount": 1000000 },
          { "denom": "atom", "amount": 40392 }
        ]
      }
    ]
  },
  "r03jf04n": {
    "gas": 343,
    "fee": { "denom": "test", "amount": 231 },
    "inputs": [
      {
        "address": "Buffer",
        "coins": [
          { "denom": "test", "amount": 1000000 },
          { "denom": "atom", "amount": 1234 }
        ],
        "sequence": 1,
        "signature": "Buffer",
        "pubkey": "PubKey"
       }
    ],
    "outputs": [
      {
        "address": "Buffer",
        "coins": [
          { "denom": "test", "amount": 1000000 },
          { "denom": "atom", "amount": 1234 }
        ]
      }
    ]
  },
  "boe9rbdi": {
    "gas": 249,
    "fee": { "denom": "test", "amount": 294 },
    "inputs": [
      {
        "address": "Buffer",
        "coins": [
          { "denom": "test", "amount": 1000000 },
          { "denom": "atom", "amount": 1234 }
        ],
        "sequence": 1,
        "signature": "Buffer",
        "pubkey": "PubKey"
       }
    ],
    "outputs": [
      {
        "address": "Buffer",
        "coins": [
          { "denom": "test", "amount": 1000000 },
          { "denom": "atom", "amount": 1234 }
        ]
      }
    ]
  }
}
