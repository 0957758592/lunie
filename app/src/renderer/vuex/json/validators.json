[
  {
    "id": "john-connor",
    "pub_key": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDZ67wzRdjbTb9HxduU9YQd9",
    "description": "Text string goes here",
    "url": "john-connor.tmp",
    "ipAddress": "35.102.58.60",
    "delegators": 7,
    "atoms": 763344,
    "startDate": "2017-05-25",
    "commission": 0.0105,
    "slashes": []
  },
  {
    "id": "agent-janssen",
    "pub_key": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDZ67wzRdjbTb9HxduU9YQd9",
    "description": "Text string goes here",
    "url": "",
    "ipAddress": "72.4.172.62",
    "delegators": 58,
    "atoms": 621242,
    "startDate": "2017-04-11",
    "commission": 0.0095,
    "slashes": []
  },
  {
    "id": "janelle-voight",
    "pub_key": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDZ67wzRdjbTb9HxduU9YQd9",
    "description": "Text string goes here",
    "url": "voigh-industries.tmp",
    "ipAddress": "23.144.70.72",
    "delegators": 33,
    "atoms": 532029,
    "startDate": "2016-12-28",
    "commission": 0.011,
    "slashes": []
  },
  {
    "id": "dr-silberman",
    "pub_key": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDZ67wzRdjbTb9HxduU9YQd9",
    "description": "Text string goes here",
    "url": "",
    "ipAddress": "209.152.16.89",
    "delegators": 17,
    "atoms": 510694,
    "startDate": "2017-05-09",
    "commission": 0.01,
    "slashes": []
  },
  {
    "id": "serena-korgan",
    "pub_key": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDZ67wzRdjbTb9HxduU9YQd9",
    "description": "Text string goes here",
    "url": "serenssimo.tmp",
    "ipAddress": "247.67.58.219",
    "delegators": 51,
    "atoms": 406938,
    "startDate": "2017-05-30",
    "commission": 0.02,
    "slashes": []
  },
  {
    "id": "kate-brewster",
    "pub_key": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDZ67wzRdjbTb9HxduU9YQd9",
    "description": "Text string goes here",
    "url": "brewster-validator.tmp",
    "ipAddress": "32.226.184.34",
    "delegators": 44,
    "atoms": 375432,
    "startDate": "2017-03-17",
    "commission": 0.09975,
    "slashes": []
  },
  {
    "id": "miles-dyson",
    "pub_key": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDZ67wzRdjbTb9HxduU9YQd9",
    "description": "Text string goes here",
    "url": "dyson-vaccums.tmp",
    "ipAddress": "252.108.145.162",
    "delegators": 38,
    "atoms": 346069,
    "startDate": "2017-02-10",
    "commission": 0.005948,
    "slashes": []
  },
  {
    "id": "marcus-wright",
    "pub_key": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDZ67wzRdjbTb9HxduU9YQd9",
    "description": "Text string goes here",
    "url": "wright-brothers.tmp",
    "ipAddress": "251.249.98.149",
    "delegators": 46,
    "atoms": 330528,
    "startDate": "2017-01-03",
    "commission": 0.0875,
    "slashes": []
  },
  {
    "id": "kyle-reese",
    "pub_key": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDZ67wzRdjbTb9HxduU9YQd9",
    "description": "Text string goes here",
    "url": "reeses-pieces.tmp",
    "ipAddress": "215.216.123.168",
    "delegators": 50,
    "atoms": 307932,
    "startDate": "2017-05-19",
    "commission": 0.0975,
    "slashes": []
  },
  {
    "id": "sarah-connor",
    "pub_key": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDZ67wzRdjbTb9HxduU9YQd9",
    "description": "Text string goes here",
    "url": "sarah-connor.tmp",
    "ipAddress": "67.164.35.159",
    "delegators": 29,
    "atoms": 306238,
    "startDate": "2017-04-13",
    "commission": 0.01125,
    "slashes": []
  },
  {
    "id": "machine-01",
    "pub_key": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDZ67wzRdjbTb9HxduU9YQd9",
    "description": "Specific note about this machine",
    "url": "",
    "ipAddress": "7.247.140.151",
    "delegators": 13,
    "atoms": 299604,
    "startDate": "2017-04-24",
    "commission": 0.0128,
    "slashes": []
  },
  {
    "id": "machine-02",
    "pub_key": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDZ67wzRdjbTb9HxduU9YQd9",
    "description": "Specific note about this machine",
    "url": "validator-machines.tmp",
    "ipAddress": "9.160.179.15",
    "delegators": 8,
    "atoms": 286434,
    "startDate": "2017-03-18",
    "commission": 0.01024,
    "slashes": []
  },
  {
    "id": "machine-03",
    "pub_key": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDZ67wzRdjbTb9HxduU9YQd9",
    "description": "Specific note about this machine",
    "url": "",
    "ipAddress": "56.183.206.90",
    "delegators": 65,
    "atoms": 270349,
    "startDate": "2017-02-23",
    "commission": 0.0808,
    "slashes": []
  },
  {
    "id": "machine-04",
    "pub_key": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDZ67wzRdjbTb9HxduU9YQd9",
    "description": "Specific note about this machine",
    "url": "",
    "ipAddress": "10.239.7.140",
    "delegators": 34,
    "atoms": 262301,
    "startDate": "2017-05-26",
    "commission": 0.0707,
    "slashes": []
  },
  {
    "id": "machine-05",
    "pub_key": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDZ67wzRdjbTb9HxduU9YQd9",
    "description": "Specific note about this machine",
    "url": "five-nines.tmp",
    "ipAddress": "139.19.4.30",
    "delegators": 68,
    "atoms": 256105,
    "startDate": "2017-04-20",
    "commission": 0.0888,
    "slashes": []
  },
  {
    "id": "t-x",
    "pub_key": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDZ67wzRdjbTb9HxduU9YQd9",
    "description": "Specific note about this machine",
    "url": "",
    "ipAddress": "236.142.154.129",
    "delegators": 45,
    "atoms": 254183,
    "startDate": "2017-04-03",
    "commission": 0.0901,
    "slashes": []
  },
  {
    "id": "guardian-01",
    "pub_key": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDZ67wzRdjbTb9HxduU9YQd9",
    "description": "Specific note about this machine",
    "url": "",
    "ipAddress": "67.27.236.231",
    "delegators": 73,
    "atoms": 235029,
    "startDate": "2017-03-29",
    "commission": 0.009,
    "slashes": []
  },
  {
    "id": "t-850",
    "pub_key": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDZ67wzRdjbTb9HxduU9YQd9",
    "description": "Specific note about this machine",
    "url": "altered-validator.tmp",
    "ipAddress": "247.248.219.156",
    "delegators": 24,
    "atoms": 183200,
    "startDate": "2017-02-16",
    "commission": 0.096,
    "slashes": []
  },
  {
    "id": "t-800",
    "pub_key": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDZ67wzRdjbTb9HxduU9YQd9",
    "description": "Specific note about this machine",
    "url": "",
    "ipAddress": "87.49.43.91",
    "delegators": 11,
    "atoms": 162395,
    "startDate": "2017-05-08",
    "commission": 0.01108,
    "slashes": []
  },
  {
    "id": "t-1000",
    "pub_key": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDZ67wzRdjbTb9HxduU9YQd9",
    "description": "Specific note about this machine",
    "url": "latest-validator.tmp",
    "ipAddress": "170.19.167.106",
    "delegators": 30,
    "atoms": 84235,
    "startDate": "2017-05-22",
    "commission": 0.094,
    "slashes": []
  }
]
