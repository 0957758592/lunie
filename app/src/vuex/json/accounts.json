{
  "difbfodb": {
    "key": "PubKeyExample",
    "sequence": 998,
    "coins": [
      { "denom": "atom", "amount": 10349 },
      { "denom": "eth", "amount": 2043.23423 },
      { "denom": "btc", "amount": 52.321 }
    ]
  }
}
