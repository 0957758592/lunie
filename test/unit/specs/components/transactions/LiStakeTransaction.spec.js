import { shallowMount } from "@vue/test-utils"
import LiStakeTransaction from "transactions/LiStakeTransaction"
import { stakingTxs } from "../../store/json/txs"

describe(`LiStakeTransaction`, () => {
  let wrapper
  const validators = [
    { operator_address: `cosmosvaloper1address1`, description: { moniker: `david` } },
    { operator_address: `cosmosvaloper1address2`, description: { moniker: `billy` } }
  ]
  const propsData = {
    transaction: stakingTxs[0],
    txType: `cosmos-sdk/MsgCreateValidator`,
    validators,
    url: `/validator`,
    bondingDenom: `stake`
  }

  beforeEach(() => {
    wrapper = shallowMount(LiStakeTransaction, {
      propsData,
      stubs: [`router-link`]
    })
  })

  it(`create validator`, () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  it(`edit validator`, async () => {
    wrapper.setProps({
      transaction: stakingTxs[1],
      txType: `cosmos-sdk/MsgDelegate`,
    })
    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  describe(`delegations`, () => {
    beforeEach(() => {
      wrapper.setProps({
        transaction: stakingTxs[2],
        txType: `cosmos-sdk/MsgDelegate`,
      })
    })

    it(`should show delegations`, () => {
      expect(wrapper.vm.$el).toMatchSnapshot()
    })

    it(`should show the validator address if no validator with that address in store`, () => {
      wrapper.setProps({
        validators: []
      })
      expect(
        wrapper
          .text()
          .indexOf(
            `To cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw`
          ) === -1
      ).toBe(true)
    })
  })

  describe(`unbonding delegations`, () => {
    beforeEach(() => {
      wrapper.setProps({
        transaction: stakingTxs[3],
        txType: `cosmos-sdk/Undelegate`,
      })
    })

    it(`should show unbondings and calculate tokens from shares`, () => {
      wrapper.setProps({ unbondingTime: Date.now() + 1000 })
      expect(wrapper.contains(`.tx-unbonding__time-diff`)).toBe(true)
      expect(wrapper.vm.$el).toMatchSnapshot()
    })

    it(`should show unbonding delegations as ended`, () => {
      wrapper.setProps({ unbondingTime: Date.now() - 1000 })
      expect(wrapper.vm.$el).toMatchSnapshot()
    })

    it(`should default to ended if no unbonding delegation is present`, () => {
      expect(wrapper.vm.$el).toMatchSnapshot()
    })
  })

  describe(`redelegations`, () => {
    it(`should show redelegations and calculate tokens from shares`, async () => {
      wrapper.setProps({
        transaction: stakingTxs[4],
        txType: `cosmos-sdk/BeginRedelegate`
      })
      await wrapper.vm.$nextTick()
      expect(wrapper.vm.$el).toMatchSnapshot()
    })
  })

  it(`unjail validator`, async () => {
    wrapper.setProps({
      transaction: stakingTxs[5],
      txType: `cosmos-sdk/MsgUnjail`,
    })
    await wrapper.vm.$nextTick()
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
