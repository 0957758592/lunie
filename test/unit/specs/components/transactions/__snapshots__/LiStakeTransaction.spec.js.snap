// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiStakeTransaction create validator 1`] = `
<li-transaction-stub
  block="193281"
  color="#512584"
>
  <div>
    
      Create validator 
    <b>
      0.1
    </b>
    <span>
       muons
    </span>
  </div>
   
  <div>
    
      Moniker: 
    <router-link-stub
      to="/validator/cosmosvaloper1qecshyc40kshszkwrtscgmsdd8tz3n4hrj9yf2"
    >
      
        cosmosvaloper1qecshyc40kshszkwrtscgmsdd8tz3n4hrj9yf2
      
    </router-link-stub>
  </div>
</li-transaction-stub>
`;

exports[`LiStakeTransaction delegations should show delegations 1`] = `
<li-transaction-stub
  block="569"
  color="#512584"
  time="42000"
>
  <div
    slot="caption"
  >
    
      Create validator 
    <b>
      NaN
    </b>
    <span>
       stakes
    </span>
  </div>
   
  <div
    slot="details"
  >
    
      Moniker: 
    <router-link-stub
      to="/validator/cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw"
    >
      
        
      
    </router-link-stub>
  </div>
</li-transaction-stub>
`;

exports[`LiStakeTransaction edit validator 1`] = `
<li-transaction-stub
  block="568"
  color="#512584"
  time="42000"
>
  <div
    slot="caption"
  >
    
      Create validator 
    <b>
      NaN
    </b>
    <span>
       stakes
    </span>
  </div>
   
  <div
    slot="details"
  >
    
      Moniker: 
    <router-link-stub
      to="/validator/cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw"
    >
      
        
      
    </router-link-stub>
  </div>
</li-transaction-stub>
`;

exports[`LiStakeTransaction redelegations should show redelegations and calculate tokens from shares 1`] = `
<li-transaction-stub
  block="193281"
  color="#512584"
>
  <div>
    
      Create validator 
    <b>
      NaN
    </b>
    <span>
       stakes
    </span>
  </div>
   
  <div>
    
      Moniker: 
    <router-link-stub
      to="/validator/undefined"
    >
      
        cosmosvaloper1qecshyc40kshszkwrtscgmsdd8tz3n4hrj9yf2
      
    </router-link-stub>
  </div>
</li-transaction-stub>
`;

exports[`LiStakeTransaction unbonding delegations should default to ended if no unbonding delegation is present 1`] = `
<div
  class="li-tx"
>
  <div
    class="li-tx__icon"
  >
    <img
      src="~assets/images/cosmos-logo.png"
      style="border-color: #512584;"
    />
  </div>
   
  <div
    class="li-tx__content"
  >
    <div
      class="li-tx__content__left"
    >
      <div
        class="li-tx__content__caption"
      >
        <p
          class="li-tx__content__caption__title"
        />
      </div>
       
      <div
        class="li-tx__content__information"
      />
    </div>
     
    <div
      class="li-tx__content__block"
    >
      <router-link-stub
        to="[object Object]"
      >
        
        Block #569 
      
      </router-link-stub>
      @ 00:00:42
    
    </div>
  </div>
</div>
`;

exports[`LiStakeTransaction unbonding delegations should show unbonding delegations as ended 1`] = `
<div
  class="li-tx"
>
  <div
    class="li-tx__icon"
  >
    <img
      src="~assets/images/cosmos-logo.png"
      style="border-color: #512584;"
    />
  </div>
   
  <div
    class="li-tx__content"
  >
    <div
      class="li-tx__content__left"
    >
      <div
        class="li-tx__content__caption"
      >
        <p
          class="li-tx__content__caption__title"
        />
      </div>
       
      <div
        class="li-tx__content__information"
      />
    </div>
     
    <div
      class="li-tx__content__block"
    >
      <router-link-stub
        to="[object Object]"
      >
        
        Block #569 
      
      </router-link-stub>
      @ 00:00:42
    
    </div>
  </div>
</div>
`;

exports[`LiStakeTransaction unjail validator 1`] = `
<li-transaction-stub
  block="193281"
  color="#512584"
>
  <div>
    
      Unjail 
    <router-link-stub
      to="/validator/cosmosvaloper1qecshyc40kshszkwrtscgmsdd8tz3n4hrj9yf2"
    >
      
        cosmosvaloper1qecshyc40kshszkwrtscgmsdd8tz3n4hrj9yf2
      
    </router-link-stub>
  </div>
</li-transaction-stub>
`;
